### ISSUE TITLE
SSL issue with feeds from Oxford University Press

### ISSUE THREAD

**Issue by msspektor** [ViennaRSS/vienna-rss#1235]
I want to subscribe to some RSS feeds of academic journals that are published by Oxford University Press. However, none of them work with Vienna (since at least version 3.5.*, cannot remember from before that), but I can access them through Google Chrome or curl. For example, here is the feed for the Journal of Consumer Research: https://academic.oup.com/rss/site_5397/3258.xml. The activity monitor outputs the following error:

```
Connecting to https://academic.oup.com/rss/site_5397/3258.xml
Error retrieving RSS feed: The network connection was lost.
```

On cocoaforge (https://forums.cocoaforge.com/viewtopic.php?f=18&t=27120), it was suggested that it might be an SSL problem:

`curl: (56) LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54`

**Comment by barijaona** [comment]
This site seems to be picky on User-Agent. Connection is reset, either on http or https, if I use curl and let it send its default user agent.

Everything works as expected if I use curl and spoof a Mozilla Firefox user agent or a Safari user agent.

````
$ curl -i https://academic.oup.com/rss/site_5397/3258.xml
curl: (56) LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 54
$ curl -i http://academic.oup.com/rss/site_5397/3258.xml
curl: (56) Recv failure: Connection reset by peer
$ curl -i http://academic.oup.com/rss/site_5397/3258.xml -A "Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:10.0) Gecko/20100101 Firefox/10.0"
HTTP/1.1 301 Moved permanently
Location: https://academic.oup.com/rss/site_5397/3258.xml
Connection: close
Cache-Control: no-cache
Pragma: no-cache
$ curl -i https://academic.oup.com/rss/site_5397/3258.xml -A "Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:10.0) Gecko/20100101 Firefox/10.0"
HTTP/1.1 200 OK
Set-Cookie: JSESSIONID=98B1402AD0780ACCB837BCFDC948838F; Path=/; HttpOnly
Accept-Ranges: bytes
Server: Microsoft-IIS/10.0
ETag: "f49df2b04c78d41:0"
Last-Modified: Fri, 09 Nov 2018 16:53:08 GMT
Date: Mon, 28 Jan 2019 11:35:43 GMT
X-SCM-Server-Number: 01
X-Powered-By: ASP.NET
Content-Type: text/xml
Content-Length: 23778
Set-Cookie: NSC_TDN_QSPE_PVQ.Xfc.THL_443=ffffffff09099b1a45525d5f4f58455e445a4a4229a0;expires=Mon, 28-Jan-2019 12:05:44 GMT;path=/;secure;httponly

````

**Comment by barijaona** [comment]
Interestingly, 
````
curl -i https://academic.oup.com/rss/site_5397/3258.xml -A "Vienna/6928 CFNetwork/902.1 Darwin/17.7.0 (x86_64)"
````
(which is very similar to the default one by NSURLSession, cf. commit b2e9b45e8f45622a8fd4eaf2071dd87435d58b2d) aborts, while
````
curl -i https://academic.oup.com/rss/site_5397/3258.xml -A "Vienna/6928 Darwin/17.7.0 (x86_64)"
````
(without the part about CFNetwork) works fine…

**Comment by barijaona** [comment]
Solved by version 3.5.4
